streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala

```diff
@@ -65,7 +65,8 @@ private[streaming] class FileBasedWriteAheadLog(
     "WriteAheadLogManager" + callerName.map(c => s" for $c").getOrElse("")
   }
   private val forkJoinPool = ThreadUtils.newForkJoinPool(threadpoolName, 20)
-  private val executionContext = ExecutionContext.fromExecutorService(forkJoinPool)
+  private val executionContext = ExecutionContext
+    .fromExecutorService(forkJoinPool, { e: Throwable => throw e })
```

Comment:
do you mean in scala 2.12 the default reporter is not throwing the exception?

Reply:
The default reporter does not throw the exception in Scala 2.11 either.

I guess there are bugs in the Future impl in Scala 2.12, since there is a big change at Future.scala in Scala 2.12.

There is a temporary fix.

I will dig into Future.scala, and hopefully propose a PR for Scala.

Reply:
Thanks @sadhen I was just going to start looking into this. This seems like a reasonable solution.

Reply:
The suggested solution will not work as intended as it will prevent subsequent callbacks (esp for other ExecutionContexts) to be submitted, esp. in the case of RejectedExecutionException.

See: 
https://github.com/scala/scala/blob/2.12.x/src/library/scala/concurrent/impl/Promise.scala#L68
https://github.com/scala/scala/blob/2.12.x/src/library/scala/concurrent/impl/Promise.scala#L284

Or, in other words, rethrowing is not "reporting" the error.

The new Future encoding included in Scala 2.13.x deals much better with RejectedExecutionExceptions and InterruptedExceptions, but it is unclear right now what parts could be backported within the binary compatibility constraints.